@here/harp-mapview/lib/text/TextElementStateCache.ts

```diff
@@ -194,4 +233,43 @@ export class TextElementStateCache {
         this.m_referenceMap.set(textElementGroup, textElementGroupState);
         this.m_sortedGroupStates = undefined;
     }
+
+    private findDuplicate(
+        elementState: TextElementState,
+        zoomLevel: number
+    ): { entry: TextElementState[]; index: number } | undefined {
```

Comment:
would it make sense to also return the duplicate? I notice there are a few cases where it writes: cacheResult.entry[cacheResult.index]; wouldn't it be a bit cleaner / simpler to just return this item?

Reply:
returning only  cacheResult.entry[cacheResult.index] won't be enough cause sometimes the caller needs to overwrite that particular entry, i.e, it needs to do:
cacheResult.entry[cacheResult.index] = newLabel;

We could return the three things: the entries array, the index and the content of the entry addressed by the index, then the object we use for the result would look like this:
`const tmpCachedDuplicate: { entries: TextElementState[]; index: number; entry:TextElementState|undefined } = {
    entries: [],
    index: -1,
    entry:undefined
};`
But then it's confusing because there's more than one way to get the same stuff, you can get `entry` also by doing `entries[index]`. Also, someone might feel tempted to do `entry = newElementState` to replace the content of an entry, which wouldn't work. And we can't even set `entry` to `readonly` cause the object is reused on every call.

Reply:
thanks for taking a look, I understand.